### PR TITLE
chore: Move http_resumer to common

### PR DESCRIPTION
**Pull request by SebOpsahl**
Due to other projects needing this change

Changelog: None

Ticket: MEN-6979

### CODE REVIEW

File: src/common/http.hpp [hunk 1]
```diff
@@ -40,13 +40,13 @@
...
-namespace update {
+namespace common {
 namespace http_resumer {
 class DownloadResumerClient;
 class HeaderHandlerFunctor;
 class BodyHandlerFunctor;
 } // namespace http_resumer
-} // namespace update
+} // namespace common
 } // namespace mender
```
Comment (lluiscampos) [inline src/common/http.hpp:50]: Too bad we still need to keep this around.

Once we are moving code, what do you think of moving `mender::common::http_resumer` namespace to `mender::common::http::resumer`? It looks a bit more tidy and it follows for example the design of `events` and `events::io`.

It however will still require that we declare here in `http.hpp` the classes of the resumer, but we can at least remove these lines and move the declarations below inside `mender::common::http` as just:
```
namespace mender {
namespace common {
namespace http {

namespace resumer {
class DownloadResumerClient;
class HeaderHandlerFunctor;
class BodyHandlerFunctor;
} // namespace resumer

using namespace std;
...
```

At this point I am only thinking out loud, it is not a requirement

File: src/mender-update/CMakeLists.txt [hunk 2]
```diff
@@ -56,7 +56,7 @@ target_sources(update_module PRIVATE
 target_compile_options(update_module PRIVATE ${PLATFORM_SPECIFIC_COMPILE_OPTIONS})
 
 add_library(mender_http_resumer STATIC
-  http_resumer/http_resumer.cpp
+  ../common/http_resumer/http_resumer.cpp
 )
 target_link_libraries(mender_http_resumer PUBLIC
   common_http
```
Comment (lluiscampos) [inline src/mender-update/CMakeLists.txt:62]: This library definition now needs to be moved to `src/common/CMakeLists.txt`.

Generally speaking we build CMake top-down approach, which means that child directories should not require files from parent directories.

File: tests/src/common/CMakeLists.txt [hunk 3]
```diff
@@ -134,3 +134,5 @@ target_link_libraries(dbus_test PUBLIC
 )
 gtest_discover_tests(dbus_test ${MENDER_TEST_FLAGS} NO_PRETTY_VALUES)
 add_dependencies(tests dbus_test)
+
+add_subdirectory(http_resumer)
```
Comment (lluiscampos) [inline tests/src/common/CMakeLists.txt:138]: Nitpick: missing newline

File: tests/src/common/http_resumer/http_resumer_test.cpp [hunk 4]
```diff
@@ -104,11 +104,12 @@ class RangeBodyOfXes : virtual public io::Reader {
...
-		// number to the power of 1.1 and round to the nearest integer. If it's odd, return
-		// 'X', if it's even, return 'x'. Due to the exponent, this pattern will change
-		// slightly throughout the sequence, giving us a chance to catch offset errors.
-		// (Note: Try printing it, the pattern is mesmerizing to watch!)
+		// Fill in a specific pattern to try to catch offset/block errors: Raise the
+		// input number to the power of 1.1 and round to the nearest integer. If
+		// it's odd, return 'X', if it's even, return 'x'. Due to the exponent, this
+		// pattern will change slightly throughout the sequence, giving us a chance
+		// to catch offset errors. (Note: Try printing it, the pattern is
+		// mesmerizing to watch!)
```
Comment (lluiscampos) [inline tests/src/common/http_resumer/http_resumer_test.cpp:112]: This file has a lot of unrelated changes. Please remove them from this PR. (If you really want to change the style of the comments, at lest do so on a separate commit).

Tip :bulb: 

I assume you got into this because your editor modifies the line with of the code and later you did `git add <file>` or even `git commit -a`. I recommend using `git add -i` (for interactive) so that you can state patch by patch which changes you want to add to your next commit :+1:

**Review by lluiscampos — CHANGES_REQUESTED** [review]
See my comments below, one slightly big suggestion for rework and some minor things to fix. Looking good otherwise!

**Review by lluiscampos — CHANGES_REQUESTED** [review]
One more thing with regards to the commit formalities:
* This change does not honor a changelog entry, so either remove it or write `Changelog: None`
    * To figure out when a change needs a changelog you need to think if the change has implications for the user. Is it a new feature or a fix: definitely has an impact on the user. Is it a rename of  a variable, moving some code, fixing a test, a typo or some other internal chore: no impact to the user hence no changelog entry
* You don't need "chore" twice in the title
* It could be shorter and more descriptive with something like: "Move `http_resumer` to common". Then in the body of the commit you can write as much as you want, giving details of the motivation (other projects can use now http_resumer bla bla bla)

:+1:

File: tests/src/common/http_resumer/http_resumer_test.cpp (hunk 4 above)
Comment (SebOpsahl) [reply tests/src/common/http_resumer/http_resumer_test.cpp]: Oh sorry, this was because the formatting tests complained about this files formatting, but must have formatted it wrongfully, will fix this. Thanks for the tip :)

File: tests/src/common/CMakeLists.txt (hunk 3 above)
Comment (SebOpsahl) [reply tests/src/common/CMakeLists.txt]: Oh, will fix it now :)

File: tests/src/common/http_resumer/http_resumer_test.cpp [hunk 5]
```diff
@@ -1271,4 +1271,4 @@ TEST_F(DownloadResumerTest, CallerDestroysReaderInTheMiddleOfAWait) {
 	// Reuse the client to check state clean-up
 	err = client->AsyncCall(req, user_header_handler, user_body_handler);
 	EXPECT_EQ(err, error::NoError) << "Unexpected error: " << err.message;
-}
+}
```
Comment (lluiscampos) [inline tests/src/common/http_resumer/http_resumer_test.cpp:1274]: Unrelated change

File: src/mender-update/daemon/context.cpp [hunk 6]
```diff
@@ -679,4 +679,4 @@ void Context::FinishDeploymentLogging() {
 
 } // namespace daemon
 } // namespace update
-} // namespace mender
+} // namespace mender
```
Comment (lluiscampos) [inline src/mender-update/daemon/context.cpp:682]: Unrelated change

File: src/common/http.hpp [hunk 7]
```diff
@@ -40,13 +40,15 @@
...
 
 namespace mender {
-namespace update {
-namespace http_resumer {
+namespace common {
+namespace http {
+namespace resumer {
 class DownloadResumerClient;
 class HeaderHandlerFunctor;
 class BodyHandlerFunctor;
```
Comment (lluiscampos) [inline src/common/http.hpp:48]: These can be moved few lines below to inside of `mender::common::http` as I tried to show you on my previous comment [here](https://github.com/mendersoftware/mender/pull/1598#discussion_r1514005108).

**Review by lluiscampos — APPROVED** [review]
Thank you for the rework. Still few nitpicks but it looks good so pre-approving :heavy_check_mark:

**Comment by SebOpsahl** [comment]
@mender-test-bot start pipeline

**Comment by SebOpsahl** [comment]
@mender-test-bot start pipeline
